src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -1100,9 +1118,24 @@ def validate_domain(self, data=None):
         Default winbind request timeout is 60 seconds, and can be adjusted by the smb4.conf parameter
         'winbind request timeout ='
         """
-        if not (await self.config())['enable']:
+        verrors = ValidationErrors()
+        config = await self.config()
+        if not config['enable']:
             return False
 
+        await self.common_validate(config, config, verrors)
+
+        try:
+            verrors.check()
+        except Exception as e:
+            await self.middleware.call(
+                'datastore.update',
+                'directoryservice.activedirectory',
+                config['id'],
+                {'ad_enable': False}
+            )
+            raise CallError(e) 
```

Comment:
@anodos325 I think you may have misunderstood me, if its validation error it seems there is no need for CallError.

Perhaps just `raise` here would do the trick?